arkouda/pdarraysetops.py

```diff
@@ -491,25 +572,45 @@ def setxor1d(pda1: pdarray, pda2: pdarray,
 
     Examples
     --------
+    Singular pdarrays
     >>> a = ak.array([1, 2, 3, 2, 4])
     >>> b = ak.array([2, 3, 5, 7, 5])
     >>> ak.setxor1d(a,b)
     array([1, 4, 5, 7])
+
+    Multiple Pdarrays Flattened
+    >>> a_vals = ak.array([1, 2, 3, 2, 4, 1, 3, 6])
+    >>> a_segs = ak.array([0, 5])
+    >>> b_vals = ak.array([2, 3, 5, 7, 5, 6, 9])
+    >>> b_segs = ak.array([0, 5])
+    >>> ak.setxor1d((a_segs, a_vals), (b_segs, b_vals))
+    (array([0, 4]), array([1, 4, 5, 7, 1, 3, 9]))
     """
-    if pda1.size == 0:
-        return pda2  # return other pdarray if pda1 is empty
-    if pda2.size == 0:
-        return pda1  # return other pdarray if pda2 is empty
-    if (pda1.dtype == int and pda2.dtype == int) or \
-            (pda1.dtype == akuint64 and pda2.dtype == akuint64):
-        repMsg = generic_msg(cmd="setxor1d", args="{} {} {}". \
-                             format(pda1.name, pda2.name, assume_unique))
-        return create_pdarray(cast(str, repMsg))
-    if not assume_unique:
-        pda1 = cast(pdarray, unique(pda1))
-        pda2 = cast(pdarray, unique(pda2))
-    aux = concatenate((pda1, pda2), ordered=False)
-    aux_sort_indices = argsort(aux)
-    aux = aux[aux_sort_indices]
-    flag = concatenate((array([True]), aux[1:] != aux[:-1], array([True])))
-    return aux[flag[1:] & flag[:-1]]
+    if isinstance(pda1, pdarray) or isinstance(pda2, pdarray):
```

Comment:
I recommend some helper methods such as for checking if two params are of the sam type